pkg/pillar/hypervisor/kvm.go

```diff
@@ -460,18 +461,134 @@ func (ctx kvmContext) Task(status *types.DomainStatus) types.Task {
 	return ctx
 }
 
-// TBD: Have a better way to calculate this number.
-// For now it is based on some trial-and-error experiments.
-// Container limit is reduced to 100MiB.
-func minVMMOverhead(config types.DomainConfig) int64 {
+func estimatedVMMOverhead(domainName string, config types.DomainConfig, aa *types.AssignableAdapters) (int64, error) {
+	// Container limit is set to 100MiB.
+	// This is a temporary solution until we have a better way to predict
+	// the memory usage of the container.
 	if config.IsOCIContainer() {
-		return 100 << 20 // Mb in bytes
+		return 100 << 20, nil // Mb in bytes
 	}
-	return 600 << 20 // Mb in bytes
+
+	var overhead int64
+
+	mmioOverhead, err := mmioVMMOverhead(domainName, config, aa)
+
+	if err != nil {
+		return 0, logError("mmioVMMOverhead() failed for domain %s: %v",
+			domainName, err)
+	}
+	overhead = undefinedVMMOverhead() + ramVMMOverhead(config) +
+		qemuVMMOverhead() + cpuVMMOverhead(config) + mmioOverhead
+
+	return overhead, nil
 }
 
-func vmmOverhead(config types.DomainConfig,
-	globalConfig *types.ConfigItemValueMap) (int64, error) {
+func ramVMMOverhead(config types.DomainConfig) int64 {
+	// 2.5% of the total RAM allocated for VM in bytes
+	// this formula is precise and well explained in the following article:
```

Comment:
If there is such article, please, put on the comments. Otherwise this comment is useless and can be removed.

Reply:
ah.. I forgot to put a link :) I put it in PR description. Fixed. And I also fixed a comment to reflect our current understanding

Reply:
for the containers, also fixed for ARM case